gobblin-iceberg/src/test/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriterTest.java

```diff
@@ -365,12 +364,6 @@ public void testFaultTolerant() throws Exception {
             new LongWatermark(52L))));
     Assert.assertEquals(gobblinMCEWriter.getDatasetErrorMap().size(), 1);
     Assert.assertEquals(gobblinMCEWriter.getDatasetErrorMap().values().iterator().next().size(), 1);
-    Assert.assertEquals(gobblinMCEWriter.getDatasetErrorMap()
-        .get(new File(tmpDir, "testDB/testTopic").getAbsolutePath())
-        .get("hivedb.testTopic").get(0).lowWatermark, 50L);
-    Assert.assertEquals(gobblinMCEWriter.getDatasetErrorMap()
-        .get(new File(tmpDir, "testDB/testTopic").getAbsolutePath())
-        .get("hivedb.testTopic").get(0).highWatermark, 52L);
```

Comment:
looks like there's still expected to be one element in the errors map, however we're no longer verifying the value.  why remove validation entirely, rather than, say, update, if need be?  now it looks like a potential oversight...

Reply:
yeah... I was also thinking the same if I could still check for the watermarks. So previously, since a single table was used and we flushed it using `gobblinMCEWriter` (in other unit test cases outside of testing completeness) before attempting `writeEnvelope` and could compare watermarks being unchanged. Now, we use a different table: `testTopicCompleteness` and writer for testing completeness: `gobblinMCEWriterWithCompletness` and the error map only contains the exception that we throw when we attempt to write a record as part of testing the fault tolerance: `Mockito.doThrow(new IOException("Test failure")).when(mockWriter).writeEnvelope(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any())`.  I have updated the assert in my latest commit to verify the content of the error map and still achieve the goal of testing fault tolerance after we've successfully flushed the metadata writer. Let me know if my understanding is correct @jack-moseley since I see you're the last person who worked on this test case!

Reply:
Yes it seems fine to me.